compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -176,9 +176,10 @@ impl IntoDiagnosticArg for hir::ConstContext {
 }
 
 /// Trait implemented by error types. This should not be implemented manually. Instead, use
-/// `#[derive(SessionSubdiagnostic)]` -- see [rustc_macros::SessionSubdiagnostic].
-#[rustc_diagnostic_item = "AddSubdiagnostic"]
-pub trait AddSubdiagnostic {
+/// `#[derive(Subdiagnostic)]` -- see [rustc_macros::Subdiagnostic].
+#[cfg_attr(bootstrap, rustc_diagnostic_item = "AddSubdiagnostic")]
+#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "Subdiagnostic")]
```

Comment:
```suggestion
#[cfg_attr(not(bootstrap), rustc_diagnostic_item = "AddToDiagnostic")]
```

Reply:
Good point, addressed [here](https://github.com/rust-lang/rust/pull/101558/commits/a6ae7f3c554bf2cf804074b4b08c506258853788).

_(Included it in the rebase/fix/etc commit)_